backend/server.py

```
@@ -49,58 +52,134 @@ def process_input_text(input_text, use_mediawiki):
 
+@app.errorhandler(HTTPException)
+def handle_http_exception(e):
+    """Return standardized JSON for HTTP exceptions."""
+    return jsonify({
+        "error": e.description,
+        "type": e.__class__.__name__,
+    }), e.code
+
+
+@app.errorhandler(Exception)
+def handle_unexpected_exception(e):
```

Ruff (0.15.2)

[warning] Unused function argument: `e`

(ARG001)